### DISCUSSION TITLE
Aqara WS-EUK03 - no state reporting after physical on/off

### DISCUSSION THREAD

**Discussion by mac2001** [Koenkk/zigbee2mqtt#16218] [Devices]
Hi, 

I saw similar discussions about Aqara switches, but am not sure if they're applicable to my problem. If so, please give me some hints. Anyway I did some research in the logs and it seems there is a problem with the newest firmware.

I've got several Aqara WS-EUK03 (lumi.switch.n1aeu1). All of them have 01-29-2021 firmware (2838) and they work perfectly fine with Zigbee2MQTT version 1.29.2-1 (as Home Assistant add on)

I have also another one (the same model), with firmware version 11-10-2021 (2839) and this one doesn't report any changes of state related to physical on/off switching (control_relay mode).
It also doesn't report power consumption or other parameters.

It's paired successfuly, and I can change it's state through software (both Z2M and HA - then the change is visible), however after changing the physical state (by pressing the switch), I don't receive ANY logs in Zigbee2MQTT console while the other switches (2838) show full communications.

Comparing the Cluster view of these two instances of switches, there are endpoint 21 and endpoint 41 missing in the problematic one.

I attach 2 files with logs - one with pairing log, the second one with manual refresh of the state (and the state itself).

Is something wrong with the device - or software 2839 changed something significantly in the switch behaviour? All working devices want to be updated to the newer version of firmware (OTA). What can I do in order to have the state (and power) reported?

Maciej

[pairing_0x54ef441000553586.txt](https://github.com/Koenkk/zigbee2mqtt/files/10429064/pairing_0x54ef441000553586.txt)
[manual_refresh_and_state_0x54ef441000553586.txt](https://github.com/Koenkk/zigbee2mqtt/files/10429065/manual_refresh_and_state_0x54ef441000553586.txt)

**Comment by SebastianOsinski** [comment]
This is really interesting as I have the same switch with the same firmware version and it works as expected - state reporting on physical on/off and power consumption work flawlessly. However, I also have two WS-EUK04s with 11-10-2021 firmware which have this problem.

I'm starting to think that it might not be only FW issue but maybe HW revision issue. 

Here is a second discussion about WS-EUK04 https://github.com/Koenkk/zigbee2mqtt/discussions/16573

**Comment by SebastianOsinski** [comment]
@mac2001 Please check how I managed to fix this issue with WS-EUK01/02/03/04 in [this discussion](https://github.com/Koenkk/zigbee2mqtt/discussions/16573#discussioncomment-4920977)

**Reply by mac2001** [reply]
Sebastian, thanks for hints. I will try to re-flash the zigbee gateway then and see if this works for me.

**Comment by mac2001** [comment]
Sebastian, 

it took me a long time, but finally I did it.

Flashing process was easy, I've used https://www.youtube.com/watch?v=JM9LnwtZRoA tutorial (link for those who are looking how to reflash the device)

So, reflashing dongle, removing device (switch) from Z2M and adding it again resolved the problem.
Thanks again!

Maciej
